src/axom/primal/examples/hex_tet_volume.cpp

```diff
@@ -0,0 +1,328 @@
+    PointType {point[0] + length, point[1], point[2]},
+    PointType {point[0], point[1] + length, point[2]},
+    PointType {point[0], point[1] + length, point[2] + length},
+    PointType {point[0] + length, point[1] + length, point[2] + length},
+    PointType {point[0] + length, point[1] + length, point[2]});
+}
+
+// Function to check intersection volumes of generated hexahedra and tetrahedra
+template <typename ExecSpace>
+void check_intersection_volumes(const Input& params)
+{
+  SLIC_INFO(axom::fmt::format(
+    "{:-^80}",
+    axom::fmt::format(
+      "Running intersection volume check in execution space: {}",
+      axom::execution_space<ExecSpace>::name())));
+
+  // Save current/default allocator
+  const int current_allocator = axom::getDefaultAllocatorID();
+
+  // Set new default allocator
+  axom::setDefaultAllocator(axom::execution_space<ExecSpace>::allocatorID());
+
+  // Generate hexahedra subdividing the unit cube with corner points
+  // (-1,-1,-1) and (1,1,1)
+  int const HEX_RESOLUTION = params.hexResolution;
+  int hex_index = 0;
+  int const NUM_HEXES = HEX_RESOLUTION * HEX_RESOLUTION * HEX_RESOLUTION;
+  HexahedronType* hexes = axom::allocate<HexahedronType>(NUM_HEXES);
+
+  SLIC_INFO(axom::fmt::format(
+    "{:-^80}",
+    axom::fmt::format(
+      "Generating {} hexahedra with hexahedra resolution set to {}",
+      NUM_HEXES,
+      HEX_RESOLUTION)));
+
+  SLIC_INFO(axom::fmt::format(
+    "{: ^80}",
+    "Hexahedra subdivide the unit cube with corner points (-1,-1,-1) "
+    "and (1,1,1)"));
+
+  for(int i = 0; i < HEX_RESOLUTION; i++)
+  {
+    for(int j = 0; j < HEX_RESOLUTION; j++)
+    {
+      for(int k = 0; k < HEX_RESOLUTION; k++)
+      {
+        double edge_length = 2.0 / HEX_RESOLUTION;
+        hexes[hex_index] = generateCube(PointType {edge_length * i - 1,
+                                                   edge_length * j - 1,
+                                                   edge_length * k - 1},
+                                        edge_length);
+        hex_index++;
+      }
+    }
+  }
+
+  // Generate tetrahedra from unit sphere with center (0,0,0)
+  int const TET_RESOLUTION = params.tetResolution;
+  int tet_index = 0;
+  int const NUM_TETS = 4 * std::pow(2, TET_RESOLUTION);
+  TetrahedronType* tets = axom::allocate<TetrahedronType>(NUM_TETS);
+  double step_size = 1.0 / std::pow(2, TET_RESOLUTION);
+
+  SLIC_INFO(axom::fmt::format(
+    "{:-^80}",
+    axom::fmt::format(
+      "Generating {} tetrahedra with tetrahedra resolution set to {}",
+      NUM_TETS,
+      TET_RESOLUTION)));
+
+  SLIC_INFO(axom::fmt::format(
+    "{: ^80}",
+    axom::fmt::format(
+      "Tetrahedra are encapsulated by the unit sphere at the origin")));
+
+  for(int i = 0; i < std::pow(2, TET_RESOLUTION); i++)
+  {
+    for(int j = 0; j <= 1; j++)
+    {
+      for(int k = 0; k <= 1; k++)
+      {
+        // Tetrahedron coordinates consist of:
+        //   - two points on the sphere where y = 0
```

Comment:
Thanks for updating the comment.
Is the following a bit clearer?

```suggestion
        //   - two points on the unit sphere great circle where y = 0
```